packages/@aws-cdk/aws-lambda/test/function.test.ts

```diff
@@ -1437,6 +1439,69 @@ describe('function', () => {
     expect(bindTarget).toEqual(fn);
   });
 
+  test('layer is baked into the function version', () => {
+    // GIVEN
+    const stack = new cdk.Stack(undefined, 'TestStack');
+    const bucket = new s3.Bucket(stack, 'Bucket');
+    const code = new lambda.S3Code(bucket, 'ObjectKey');
+
+    const fn = new lambda.Function(stack, 'fn', {
+      runtime: lambda.Runtime.NODEJS_14_X,
+      code: lambda.Code.fromInline('exports.main = function() { console.log("DONE"); }'),
+      handler: 'index.main',
+    });
+
+    const fnHash = calculateFunctionHash(fn);
+
+    // WHEN
+    const layer = new lambda.LayerVersion(stack, 'LayerVersion', {
+      code,
+      compatibleRuntimes: [lambda.Runtime.NODEJS_14_X],
+    });
+
+    fn.addLayers(layer);
+
+    const newFnHash = calculateFunctionHash(fn);
+
+    expect(fnHash).not.toEqual(newFnHash);
+  });
+
+  test('with feature flag, layer version is baked into function version', () => {
```

Comment:
This is the most important test that should validate that the bug is fixed. It is hard to actually model the process described in the issue, but I hope this is sufficient. My idea is that specifically, the `s3Key` will change when the layer code is updated, thereby changing the layershash. Curious if I am missing any considerations like if the s3Key is a token.

Reply:
Yeah, it's not going to work when `s3Key` is a token.